nixos/modules/services/desktops/pipewire/pipewire-media-session.nix

```diff
@@ -0,0 +1,340 @@
+# pipewire example session manager.
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pipewire.media-session;
+  enable32BitAlsaPlugins = cfg.alsa.support32Bit
+                           && pkgs.stdenv.isx86_64
+                           && pkgs.pkgsi686Linux.pipewire != null;
+
+  # Helpers for generating the pipewire JSON config file
+  mkSPAValueString = v:
+  if builtins.isList v then "[${lib.concatMapStringsSep " " mkSPAValueString v}]"
+  else if lib.types.attrs.check v then
+    "{${lib.concatStringsSep " " (mkSPAKeyValue v)}}"
+  else lib.generators.mkValueStringDefault { } v;
+
+  mkSPAKeyValue = attrs: map (def: def.content) (
+  lib.sortProperties
+    (
+      lib.mapAttrsToList
+        (k: v: lib.mkOrder (v._priority or 1000) "${lib.escape [ "=" ] k} = ${mkSPAValueString (v._content or v)}")
+        attrs
+    )
+  );
+
+  toSPAJSON = attrs: lib.concatStringsSep "\n" (mkSPAKeyValue attrs);
+in {
+
+  meta = {
+    maintainers = teams.freedesktop.members;
+  };
+
+  ###### interface
+  options = {
+    services.pipewire.media-session = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
```

Comment:
@jansol this seems like a bad idea. i just ran into a problem when building a system and pipewire being pulled in due to this

Reply:
I'm open to better suggestions / instructions. The idea was that since pipewire-media-session is currently the only packaged session manager it has to be enabled by default. Of course it is not relevant when pipewire itself is not enabled.

Reply:
Would it be possible to set the default to `config.services.pipewire.enabled`? Since it wouldn't be relevant otherwise as you said.

Reply:
@minijackson suggestion sounds reasonable to me. that would prevent pulling in pipewire into a system when not even being used.

Reply:
fixed in https://github.com/NixOS/nixpkgs/pull/113772